httpcore/dispatch/connection.py

```diff
@@ -38,7 +38,12 @@ def __init__(
         self.h2_connection = None  # type: typing.Optional[HTTP2Connection]
 
     def prepare_request(self, request: Request) -> None:
-        request.prepare()
+        if self.h11_connection is not None:
+            self.h11_connection.prepare_request(request)
+        elif self.h2_connection is not None:
+            self.h2_connection.prepare_request(request)
+        else:
+            request.prepare()
```

Comment:
We don't want this change.
I'll likely switch some things around so that the H11/H2 don't have the dummy `prepare_request` methods in the first place.